homeassistant/components/motionmount/select.py

```diff
@@ -72,8 +88,9 @@ def current_option(self) -> str | None:
     async def async_select_option(self, option: str) -> None:
         """Set the new option."""
         index = int(option[:1])
-        await self.mm.go_to_preset(index)
-        self._attr_current_option = option
-
-        # Perform an update so we detect changes to the presets (changes are not pushed)
-        self.async_schedule_update_ha_state(True)
+        try:
+            await self.mm.go_to_preset(index)
+        except (TimeoutError, socket.gaierror) as ex:
+            raise HomeAssistantError("Failed to communicate with MotionMount") from ex
```

Comment:
For a followup, you can add translations to the exceptions you raise to enhance the non native english speaker experience